src/Umbraco.Infrastructure/PropertyEditors/MultiUrlPickerValueEditor.cs

```diff
@@ -138,6 +140,7 @@ public IEnumerable<UmbracoEntityReference> GetReferences(object? value)
                         udi = new GuidUdi(Constants.UdiEntityType.Document, documentEntity.Key);
                         url = publishedSnapshot.Content?.GetById(entity.Key)?.Url(_publishedUrlProvider) ?? "#";
                         trashed = documentEntity.Trashed;
+                        content = publishedSnapshot.Content?.GetById(entity.Key);
```

Comment:
Changing the order here can avoid fetching the `IPublishedContent` from the cache twice:
```suggestion
                        trashed = documentEntity.Trashed;
                        content = publishedSnapshot.Content?.GetById(entity.Key);
                        url = content?.Url(_publishedUrlProvider) ?? "#";
```